File: com.issue.tracker.infra.persistence/src/main/java/com/issue/tracker/infra/persistence/common/BaseEntity.java
package com.issue.tracker.infra.persistence.common;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

import java.io.Serializable;

@MappedSuperclass
public abstract class BaseEntity<ID extends Serializable> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected ID id;

    public BaseEntity() {

    }

    public BaseEntity(ID id) {
        this.id = id;
    }

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }
}

File: com.issue.tracker.infra.persistence/src/main/kotlin/com/issue/tracker/infra/persistence/common/BaseEntity.kt
package com.issue.tracker.infra.persistence.common

import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.io.Serializable

@MappedSuperclass
abstract class BaseEntity<ID : Serializable>(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: ID? = null
)